CraftTweaker2-MC1120-Main/src/main/java/crafttweaker/mc1120/block/MCBlockState.java

```diff
@@ -51,5 +56,38 @@ public int compare(crafttweaker.api.block.IBlockState other) {
         }
         return result;
     }
-    
+
+    @Override
+    public crafttweaker.api.block.IBlockState withProperty(String name, String value) {
+        IProperty property = blockState.getBlock().getBlockState().getProperty(name);
+        //noinspection unchecked
+        Optional<? extends Comparable> propValue = property.parseValue(value);
+        if (propValue.isPresent()) {
+            //noinspection unchecked
+            return new MCBlockState(blockState.withProperty(property, propValue.get()));
+        }
+        return null;

```

Comment:
I'm not sure if return null in ZS is that great. Perhaps just return current state and log a message of some sort? At least Log a message.